### DISCUSSION TITLE
List all dataset variables with print()

### DISCUSSION THREAD

**Discussion by st-bender** [pydata/xarray#5535] [General] [answered]
Hi,

It looks like version 0.17 broke the console display of variables in a dataset.
Version 0.16:
```
In [4]: print(ds)
Out[4]: 
<xarray.Dataset>
Dimensions:  (alt: 1, lat: 1, lon: 1, time: 1)
Coordinates:
  * time     (time) datetime64[ns] 2009-06-21T08:03:20
  * alt      (alt) float64 10.0
  * lat      (lat) float64 20.0
  * lon      (lon) float64 30.0
Data variables:
    He       (time, alt, lat, lon) float64 4.568e+13
    O        (time, alt, lat, lon) float64 0.0
    N2       (time, alt, lat, lon) float64 6.807e+18
    O2       (time, alt, lat, lon) float64 1.826e+18
    Ar       (time, alt, lat, lon) float64 8.142e+16
    rho      (time, alt, lat, lon) float64 0.0004188
    H        (time, alt, lat, lon) float64 0.0
    N        (time, alt, lat, lon) float64 0.0
    AnomO    (time, alt, lat, lon) float64 0.0
    Texo     (time, alt, lat, lon) float64 1.027e+03
    Talt     (time, alt, lat, lon) float64 237.3
    lst      (time, lon) float64 10.06
    Ap       (time) int32 6
    f107     (time) float64 66.7
    f107a    (time) float64 69.0
```

Version 0.17:
```
In [27]: print(ds)
<xarray.Dataset>
Dimensions:  (alt: 1, lat: 1, lon: 1, time: 1)
Coordinates:
  * time     (time) datetime64[ns] 2009-06-21T08:03:20
  * alt      (alt) float64 10.0
  * lat      (lat) float64 20.0
  * lon      (lon) float64 30.0
Data variables: (12/15)
    He       (time, alt, lat, lon) float64 4.568e+13
    O        (time, alt, lat, lon) float64 0.0
    N2       (time, alt, lat, lon) float64 6.807e+18
    O2       (time, alt, lat, lon) float64 1.826e+18
    Ar       (time, alt, lat, lon) float64 8.142e+16
    rho      (time, alt, lat, lon) float64 0.0004188
    ...       ...
    Texo     (time, alt, lat, lon) float64 1.027e+03
    Talt     (time, alt, lat, lon) float64 237.3
    lst      (time, lon) float64 10.06
    Ap       (time) int32 6
    f107     (time) float64 66.7
    f107a    (time) float64 69.0
```

This new behaviour breaks all my doctests. How can I get the old one back?

These frequent interface changes are really annoying, please stop doing that to people. 

Cheers.

**Comment by dcherian** [comment, marked as answer]
I think you can control this by setting `display_max_rows`: https://xarray.pydata.org/en/stable/generated/xarray.set_options.html

**Reply by Illviljan** [reply]
If you want to change the doctests results you can try out: https://github.com/max-sixty/pytest-accept

**Reply by max-sixty** [reply]
@st-bender I appreciate the changing reprs are frustrating, and are "taxing the wise" who run doctests. We will try and mitigate them. Sometimes it does become important though — e.g. we'll find something that prints far too much.

Thanks for @Illviljan 's suggestion — I wrote that pytest plugin for exactly that reason. If it doesn't work for you, post an issue and I'll take a look

**Reply by st-bender** [reply]
@max-sixty I tend to disagree, I frequently use(d) `print(ds)` or just `ds` at the console or in a notebook to have a quick look that **all** the variables ended up correctly in the dataset, be it by reading a netcdf or some other construction. Stripping variables from that output is not a good idea imo, and I would appreciate it if the default limit were much higher if it really is a problem that it prints too much. Was it a problem in the first place? Don't fix it if it isn't broken.

Thanks to @dcherian for pointing out the solution, strangely it didn't show up with `xr.set_options?` ?
However, I decided to pin the xarray version for ci, since that's basically the only place where it mattered, and `xr.set_options()` would most probably fail with earlier xarray versions.

**Reply by max-sixty** [reply]
A PR is welcome for the `xr.set_options` docs if you'd like to contribute @st-bender
